### ISSUE TITLE
WebView2 support for IE7 or IE9

### ISSUE THREAD

**Issue by dealsweb** [MicrosoftEdge/WebView2Feedback#440]
Does WebView2 still support internet explorer rendering? I have a legacy application so it is still dependent on IE7/9 rendering in some cases and currently using WebBrowser control. 
I couldn't find any documentation on WebView2 that supports IE versions.

[AB#29384395](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/29384395)

**Comment by champnic** [comment]
We don't currently support using the IE rendering engine (trident) for older pages. If that's a strong requirement, I would suggest continuing to use the WebBrowser control. You could choose to use both, and switch between them depending on the page (this is similar to what Edge's IE Mode does today). I'll add this to our backlog, but I wouldn't expect support here for a long time. Thanks!

**Comment by RickStrahl** [comment]
Please no - lets not make this control more complex than it already is by including legacy functionality :smile:

@champnic  seems reasonable if that sort of backwards compatibility is needed.

**Comment by gg1975developer** [comment]
Hi, we are using Activex InternetExplorer.Application. Will it be available also after the dismission of ie in the 2022? Will it be available the WebBrowser Activex?

**Comment by champnic** [comment]
@gg1975developer Yes, WebBrowser will continue to be supported after IE is removed.

**Comment by gg1975developer** [comment]
Hi perfect thanks. Both of them (InternetExplorer.Application and WebBrowser) or just WebBrowser will be supported? They are two different activex as far as I know. I'm trying to figure out also how to pass cookies, postdata to the webbrowser component from Edge. I'm trying to use add_WebResourceRequested to catch them. But I don't know if could be a solution. Thanks

**Comment by schikla** [comment]
> @gg1975developer Yes, WebBrowser will continue to be supported after IE is removed.

Where can I find an official documentation to that statement?

**Comment by champnic** [comment]
https://techcommunity.microsoft.com/t5/windows-it-pro-blog/internet-explorer-11-desktop-app-retirement-faq/ba-p/2366549

> Out of scope at the time of this announcement (unaffected):
> 
> Internet Explorer mode in Microsoft Edge
> Internet Explorer platform (MSHTML/Trident), including WebOC

> Technical details
> What is the MSHTML (Trident) engine? How does that relate to IE mode? And what about WebOC? Or the MSHTA app?
> 
> The MSHTML (Trident) engine is the underlying platform for Internet Explorer 11. This is the same engine used by IE mode and it will continue to be supported (in other words, unaffected by this announcement). Both WebOC and the MSHTA app will continue to be supported as they rely on the MSHTML engine which is unaffected by this announcement. If you have a custom or third-party app that relies on the MSHTML platform, you can expect it to continue to work. For future app development, we recommend [using WebView2](https://docs.microsoft.com/microsoft-edge/webview2/).

**Comment by liminzhu** [comment]
Hi folks. On WebView2 side, it is very unlikely we will ever add support for IE on the WebView2 side. Closing this.
